sql/log_event.cc

```diff
@@ -5216,6 +5216,13 @@ int Query_log_event::do_apply_event(Relay_log_info const *rli,
   thd->m_statement_psi= NULL;
   thd->m_digest= NULL;
 
+  /*
+    Prevent rewritten query from getting "stuck" in SHOW PROCESSLIST,
+    and performance_schema.threads.
+  */
+  thd->rewritten_query.mem_free();
```

Comment:
should you chcek if rewritten_query is used first?
```suggestion
 if (thd->rewritten_query.length())
   thd->rewritten_query.mem_free();
 ```

Reply:
On master size, we always free it - https://github.com/percona/percona-server/blob/8.0/sql/sql_parse.cc#L2262

Reply:
Thanks for confirming